### PR TITLE
Feat/ci 62/implement card order sidebar creation step

### PR DESCRIPTION
**Pull request by julianromero-bc**
## Description
Add get product request for obtaining the available type of cards the user could create
Add create card request
Add the flow of creating a card in the card order sidebar. 
Working states:
- Loading
- Failure

## Demo
debit-card-sidebar-create-loading-fail-status

### CODE REVIEW

File: packages/blockchain-wallet-v4-frontend/src/modals/OrderMyCard/index.tsx [hunk 1]
```diff
@@ -1,34 +1,77 @@
...
+      Failure: () => <FailedCreationStep handleClose={handleClose} handleRetry={handleRetry} />,
+      Loading: () => (
+        <LoadingWrapper>
+          <Loading text={LoadingTextEnum.PROCESSING} />
+        </LoadingWrapper>
+      ),
+      NotAsked: () => (
+        <CreateCardStep handleClose={handleClose} handleCreateCard={handleCreateCard} />
+      ),
+      Success: () => <>Success</>
```
Comment (julianromero-bc) [inline packages/blockchain-wallet-v4-frontend/src/modals/OrderMyCard/index.tsx:55]: In next PR I will add the view and continue the flow of Success

File: packages/blockchain-wallet-v4-frontend/src/modals/OrderMyCard/index.tsx [hunk 2]
```diff
@@ -1,34 +1,77 @@
...
 import CreateCardStep from './CreateCardStep'
+import FailedCreationStep from './FailedCreationStep'
+import { LoadingWrapper } from './model'
+import { getData } from './selectors'
 
 const OrderMyCard = (props: Props) => {
   const [show, setShow] = useState(true)
 
+  // MVP only one product will be available, for V1 there will be a product selector in the UI
+  const { productCode } = props.products[0]
```
Comment (blockdylanb) [inline packages/blockchain-wallet-v4-frontend/src/modals/OrderMyCard/index.tsx:22]: you could double destructure this like `const [{ productCode }] = props.products` but it's not a blocker.

File: packages/blockchain-wallet-v4-frontend/src/modals/OrderMyCard/FailedCreationStep/index.tsx [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+import { ResultSubTitleWrapper, ResultTitleWrapper, ResultWrapper, StyledButton } from '../model'
+
+type Props = {
+  handleClose: () => void
+  handleRetry: () => void
+}
+
+const FailedCreationStep = ({ handleClose, handleRetry }: Props) => (
+  <>
```
Comment (blockdylanb) [inline packages/blockchain-wallet-v4-frontend/src/modals/OrderMyCard/FailedCreationStep/index.tsx:16]: Could you import `FlyoutContainer` and use that here? Will just be a bit cleaner and future proof this modal if we ever change flyouts in the future.

**Review by blockdylanb — CHANGES_REQUESTED** [review]
Looks great overall, mind just fixing the FlyoutContainer issue?

File: packages/blockchain-wallet-v4-frontend/src/modals/OrderMyCard/index.tsx (hunk 2 above)
Comment (julianromero-bc) [reply packages/blockchain-wallet-v4-frontend/src/modals/OrderMyCard/index.tsx]: good catch. Done

File: packages/blockchain-wallet-v4-frontend/src/modals/OrderMyCard/FailedCreationStep/index.tsx (hunk 3 above)
Comment (julianromero-bc) [reply packages/blockchain-wallet-v4-frontend/src/modals/OrderMyCard/FailedCreationStep/index.tsx]: Thanks, I added that and I remove an unnecessary styling

File: .github/workflows/deploy-storybook.yml [hunk 4]
```diff
@@ -24,8 +24,9 @@ jobs:
       - name: Install dependencies
         run: yarn
 
-      - name: Deploy storybook - base components
-        run: yarn storybook:base
+#      Not working command
```
Comment (pedroapfilho) [inline .github/workflows/deploy-storybook.yml:27]: Could you do this in a different PR?

File: packages/blockchain-wallet-v4-frontend/src/data/components/actions.ts [hunk 5]
```diff
@@ -3,6 +3,7 @@ import { actions as brokerage } from './brokerage/slice'
 import * as btcTransactions from './btcTransactions/actions'
 import { actions as buySell } from './buySell/slice'
 import * as coinTransactions from './coinTransactions/actions'
+import { actions as debitCard } from './debitCard/slice'
```
Comment (pedroapfilho) [inline packages/blockchain-wallet-v4-frontend/src/data/components/actions.ts:6]: nice one!

File: packages/blockchain-wallet-v4-frontend/src/data/components/debitCard/slice.ts [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+    },
+    getProducts: () => {},
+    getProductsFailure: (state, action) => {
+      state.products = action.payload
+    },
+    getProductsSuccess: (state, action) => {
+      state.products = action.payload
+    },
+    resetCreateCardState: (state) => {
+      state.cardCreationData = Remote.NotAsked
```
Comment (pedroapfilho) [inline packages/blockchain-wallet-v4-frontend/src/data/components/debitCard/slice.ts:34]: very nice job!

File: packages/blockchain-wallet-v4-frontend/src/modals/OrderMyCard/CreateCardStep/index.tsx [hunk 7]
```diff
@@ -37,7 +37,7 @@ const CreateCardStep = ({ handleClose, handleCreateCard }: Props) => (
       {/* disabled for MVP, will need this for V1 */}
       {/* <SeeCardDetailsBtn onClick={handleShowCardDetails} /> */}
 
-      <Image name='order-my-card-icon' />
+      <Image name='order-my-card-icon' style={{ marginBottom: '32px' }} />
```
Comment (pedroapfilho) [inline packages/blockchain-wallet-v4-frontend/src/modals/OrderMyCard/CreateCardStep/index.tsx:40]: `styled-component`?

File: packages/blockchain-wallet-v4-frontend/src/modals/OrderMyCard/index.tsx (hunk 1 above)
Comment (pedroapfilho) [reply packages/blockchain-wallet-v4-frontend/src/modals/OrderMyCard/index.tsx:55]: so could you add a `// TODO` here?

File: packages/blockchain-wallet-v4-frontend/src/modals/OrderMyCard/model.tsx [hunk 8]
```diff
@@ -28,9 +28,33 @@ const StyledSeeCardDetailsButton = styled(Button)`
...
-  margin-top: 32px;
+export const StyledButton = styled(Button)`
   width: 327px;
+  margin-bottom: 8px;
+`
+
+export const ResultWrapper = styled.div`
+  width: 327px;
+  text-align: center;
+  margin-top: -10%;
```
Comment (pedroapfilho) [inline packages/blockchain-wallet-v4-frontend/src/modals/OrderMyCard/model.tsx:39]: dangerous. Could you explain why did you do this?

File: packages/blockchain-wallet-v4-frontend/src/modals/OrderMyCard/model.tsx [hunk 9]
```diff
@@ -28,9 +28,33 @@ const StyledSeeCardDetailsButton = styled(Button)`
...
+  color: black;
+  font-weight: 600;
+  margin-bottom: 8px;
+`
+
+export const ResultSubTitleWrapper = styled(Text)`
+  font-size: 0.875rem;
+  color: ${(props) => props.theme.grey600};
+  font-weight: 500;
+  margin-bottom: 24px;
```
Comment (pedroapfilho) [inline packages/blockchain-wallet-v4-frontend/src/modals/OrderMyCard/model.tsx:53]: rem?

File: packages/blockchain-wallet-v4-frontend/src/scenes/DebitCard/template.tsx [hunk 10]
```diff
@@ -51,7 +51,11 @@ const LinkHereButton = () => (
   </div>
 )
 
-const DebitCard = ({ modalActions }: Props) => {
+const DebitCard = ({ debitCardActions, modalActions }: Props) => {
+  useEffect(() => {
+    // Get available products that user can create
```
Comment (pedroapfilho) [inline packages/blockchain-wallet-v4-frontend/src/scenes/DebitCard/template.tsx:56]: can't this be on the `mapStateToProps`?

File: packages/blockchain-wallet-v4/src/network/api/debitCard/index.ts [hunk 11]
```diff
@@ -0,0 +1,24 @@
+import { ProductType } from 'data/components/debitCard/types'
+
+export default ({ authorizedGet, authorizedPost, nabuUrl }) => {
+  const createDCCard = (productCode: string): string =>
```
Comment (pedroapfilho) [inline packages/blockchain-wallet-v4/src/network/api/debitCard/index.ts:4]: what does it mean? `Create Debit Card Card`?

**Review by pedroapfilho — CHANGES_REQUESTED** [review]
Very nice work! Just added a few comments, but overall, it's very good!

File: packages/blockchain-wallet-v4-frontend/src/modals/OrderMyCard/CreateCardStep/index.tsx (hunk 7 above)
Comment (julianromero-bc) [reply packages/blockchain-wallet-v4-frontend/src/modals/OrderMyCard/CreateCardStep/index.tsx:40]: I prefer to leave just the style here instead of adding more lines for a simple margin

File: packages/blockchain-wallet-v4-frontend/src/modals/OrderMyCard/model.tsx (hunk 9 above)
Comment (julianromero-bc) [reply packages/blockchain-wallet-v4-frontend/src/modals/OrderMyCard/model.tsx:53]: In other PR where I put rem I was asked to use px, while there is no clear definition about rem or px in margins I would prefer to keep it simple with px

File: .github/workflows/deploy-storybook.yml (hunk 4 above)
Comment (julianromero-bc) [reply .github/workflows/deploy-storybook.yml:27]: Sure, removed

File: packages/blockchain-wallet-v4-frontend/src/scenes/DebitCard/template.tsx (hunk 10 above)
Comment (julianromero-bc) [reply packages/blockchain-wallet-v4-frontend/src/scenes/DebitCard/template.tsx:56]: I prefer to leave this here, for now, this might change in the following PRs and if I add to mapStateToProps I would also have to add to pass it through the props to this component. Adding a lot of code lines unnecessary

File: packages/blockchain-wallet-v4/src/network/api/debitCard/index.ts (hunk 11 above)
Comment (julianromero-bc) [reply packages/blockchain-wallet-v4/src/network/api/debitCard/index.ts:4]: good catch, removed second card

File: packages/blockchain-wallet-v4-frontend/src/modals/OrderMyCard/CreateCardStep/index.tsx (hunk 7 above)
Comment (pedroapfilho) [reply packages/blockchain-wallet-v4-frontend/src/modals/OrderMyCard/CreateCardStep/index.tsx:40]: Sure, but what if I add you to add margin top of 12px?

File: packages/blockchain-wallet-v4-frontend/src/scenes/DebitCard/template.tsx (hunk 10 above)
Comment (pedroapfilho) [reply packages/blockchain-wallet-v4-frontend/src/scenes/DebitCard/template.tsx:56]: Do you see this pattern somewhere else?

Comment (julianromero-bc) [reply packages/blockchain-wallet-v4-frontend/src/scenes/DebitCard/template.tsx:56]: Im sorry I don’t understand what you are trying to say

File: packages/blockchain-wallet-v4-frontend/src/modals/OrderMyCard/CreateCardStep/index.tsx (hunk 7 above)
Comment (julianromero-bc) [reply packages/blockchain-wallet-v4-frontend/src/modals/OrderMyCard/CreateCardStep/index.tsx:40]: I could add it to the “SubTexWrapper” that is up above

**Review by pedroapfilho — APPROVED** [review]

**Review by blockdylanb — APPROVED** [review]

File: packages/blockchain-wallet-v4-frontend/src/scenes/DebitCard/template.tsx (hunk 10 above)
Comment (blockdylanb) [reply packages/blockchain-wallet-v4-frontend/src/scenes/DebitCard/template.tsx:56]: @pedroapfilho do you mean add `debitCardActions.getProducts()` to mapDispatchToProps? mapStateToProps is only really for selectors, not actions.
